nifi-extension-bundles/nifi-aws-bundle/nifi-aws-kinesis/src/main/java/org/apache/nifi/processors/aws/kinesis/ConsumeKinesis.java

```diff
@@ -0,0 +1,652 @@
+        @WritesAttribute(attribute = MIME_TYPE,
+                description = "Sets the mime.type attribute to the MIME Type specified by the Record Writer (if configured)"),
+        @WritesAttribute(attribute = RECORD_COUNT,
+                description = "Number of records written to the FlowFiles by the Record Writer (if configured)"),
+        @WritesAttribute(attribute = RECORD_ERROR_MESSAGE,
+                description = "This attribute provides on failure the error message encountered by the Record Reader or Record Writer (if configured)")
+})
+@DefaultSettings(yieldDuration = "100 millis")
+@SystemResourceConsideration(resource = SystemResource.CPU, description = """
+        Kinesis Client Library is used to create a Worker thread for consumption of Kinesis Records.
+        The Worker is initialised and started when this Processor has been scheduled. It runs continually, spawning Kinesis Record Processors as required \
+        to fetch Kinesis Records. The Worker Thread (and any child Record Processor threads) are not controlled by the normal NiFi scheduler as part of the \
+        Concurrent Thread pool and are not released until this processor is stopped.""")
+@SystemResourceConsideration(resource = SystemResource.NETWORK, description = """
+        Kinesis Client Library will continually poll for new Records, \
+        requesting up to a maximum number of Records/bytes per call. This can result in sustained network usage.""")
+@SystemResourceConsideration(resource = SystemResource.MEMORY, description = """
+        ConsumeKinesis buffers Kinesis Records in memory until they can be processed.
+        The maximum size of the buffer is controlled by the 'Max Bytes to Buffer' property.
+        In addition, Kinesis Client Library may cache some amount of data for each shard when the processor's buffer is full.""")
+public class ConsumeKinesis extends AbstractProcessor {
+
+    private static final Duration HTTP_CLIENTS_CONNECTION_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration HTTP_CLIENTS_READ_TIMEOUT = Duration.ofMinutes(3);
+
+    /**
+     * Best balance between throughput and CPU usage by KCL.
+     */
+    private static final int KINESIS_HTTP_CLIENT_CONCURRENCY_PER_TASK = 16;
+    private static final int KINESIS_HTTP_CLIENT_WINDOW_SIZE_BYTES = 512 * 1024; // 512 KiB
+    private static final Duration KINESIS_HTTP_HEALTH_CHECK_PERIOD = Duration.ofMinutes(1);
+
+    static final PropertyDescriptor KINESIS_STREAM_NAME = new PropertyDescriptor.Builder()
+            .name("Amazon Kinesis Stream Name")
+            .description("The name of the Kinesis stream to consume from.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor APPLICATION_NAME = new PropertyDescriptor.Builder()
+            .name("Application Name")
```

Comment:
Does this need to be a configurable property? Automatically setting the value to `ConsumeKinesis` plus the component identifier would produce a unique value for each Processor. Alternatively, keeping the property enables reusing an existing reference, is that needed?